### ISSUE TITLE
Help with study case - Low evaluation scores

### ISSUE THREAD

**Issue by MayaGueguen** [biomodhub/biomod2#304]
Dear @rpatin,

Today I am doing a bit different modelization: I am building this model over Europe considering the 19 WolrdClim variables and using more than 3'500 points of presence. However, I am still facing the same problems with the `randomForest` as well as pretty bad evaluation scores for the other algorithms. I really hope you can help me out.

Hereafter, I show you my data, the modeling steps and the results I get so far.

```r
Capreolus_BIOMOD_data <- BIOMOD_FormatingData(
  resp.name = "Capreolus Capreolus",
  resp.var = Capreolus_points,
  expl.var = Bioclim_Capreolus,
  PA.nb.rep = 1,
  PA.nb.absences = 20000,
  PA.strategy = 'random'
)
Capreolus_BIOMOD_data

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-= BIOMOD.formated.data -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

dir.name =  .

sp.name =  Capreolus.Capreolus

	 3064 presences,  0 true absences and  20000 undefined points in dataset


	 7 explanatory variables

     bio_2            bio_7            bio_10          bio_11           bio_12           bio_14      
 Min.   : 1.851   Min.   : 4.496   Min.   :  6.0   Min.   :  25.0   Min.   : 3.220   Min.   : 273.9  
 1st Qu.:14.422   1st Qu.:24.001   1st Qu.:169.0   1st Qu.: 102.0   1st Qu.: 7.824   1st Qu.: 665.7  
 Median :16.426   Median :30.488   Median :203.0   Median : 119.0   Median : 8.464   Median : 824.1  
 Mean   :16.585   Mean   :30.349   Mean   :197.7   Mean   : 154.8   Mean   : 8.591   Mean   : 837.1  
 3rd Qu.:18.714   3rd Qu.:34.727   3rd Qu.:226.0   3rd Qu.: 173.0   3rd Qu.: 9.355   3rd Qu.:1002.4  
 Max.   :27.113   Max.   :99.578   Max.   :659.0   Max.   :1148.0   Max.   :15.065   Max.   :1414.0  
     bio_18      
 Min.   :-9.449  
 1st Qu.: 9.669  
 Median :14.036  
 Mean   :12.933  
 3rd Qu.:16.842  
 Max.   :23.394  


 1 Pseudo Absences dataset available ( PA1 ) with  20000 (PA1) pseudo absences

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

myBIOMODoption <- BIOMOD_ModelingOptions()

Capreolus_single_models <- BIOMOD_Modeling(
  bm.format = Capreolus_BIOMOD_data,
  modeling.id = "Single_models",
  models = c("GLM", "GBM", "GAM", "RF", "MAXENT", "MAXNET"),
  bm.options = myBIOMODoption,
  CV.strategy = "random",
  CV.nb.rep = 10,
  CV.perc = 0.7,
  CV.do.full.models = F,
  metric.eval = c("ROC", "TSS"),
  var.import = 1,
  nb.cpu = 4,
  seed.val = 42,
  do.progress = T)

scores <- get_evaluations(Capreolus_single_models)
scores

``` 

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Loren/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Loren/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

  | full.name | PA | run | algo | metric.eval | cutoff | sensitivity | specificity | calibration | validation | evaluation
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | Capreolus.Capreolus_PA1_RUN1_GLM | PA1 | RUN1 | GLM | ROC | 496,5 | 81,725 | 55,757 | 0,744 | 0,749 | NA
2 | Capreolus.Capreolus_PA1_RUN1_GLM | PA1 | RUN1 | GLM | TSS | 498 | 81,538 | 55,85 | 0,373 | 0,386 | NA
3 | Capreolus.Capreolus_PA1_RUN1_GBM | PA1 | RUN1 | GBM | ROC | 550,5 | 77,855 | 63,1 | 0,76 | 0,756 | NA
4 | Capreolus.Capreolus_PA1_RUN1_GBM | PA1 | RUN1 | GBM | TSS | 550 | 78,182 | 62,757 | 0,41 | 0,403 | NA
5 | Capreolus.Capreolus_PA1_RUN1_GAM | PA1 | RUN1 | GAM | ROC | 463,5 | 84,709 | 54,493 | 0,752 | 0,754 | NA
6 | Capreolus.Capreolus_PA1_RUN1_GAM | PA1 | RUN1 | GAM | TSS | 459 | 85,128 | 53,936 | 0,391 | 0,385 | NA
7 | Capreolus.Capreolus_PA1_RUN1_RF | PA1 | RUN1 | RF | ROC | 370 | 99,907 | 99,943 | 1 | 0,728 | NA
8 | Capreolus.Capreolus_PA1_RUN1_RF | PA1 | RUN1 | RF | TSS | 330 | 100 | 99,836 | 0,998 | 0,048 | NA
9 | Capreolus.Capreolus_PA1_RUN1_MAXENT | PA1 | RUN1 | MAXENT | ROC | 434,5 | 76,643 | 63,029 | 0,762 | 0,755 | NA
10 | Capreolus.Capreolus_PA1_RUN1_MAXENT | PA1 | RUN1 | MAXENT | TSS | 445 | 75,058 | 64,514 | 0,396 | 0,402 | NA
11 | Capreolus.Capreolus_PA1_RUN1_MAXNET | PA1 | RUN1 | MAXNET | ROC | 406,5 | 81,818 | 56,421 | 0,75 | 0,753 | NA
12 | Capreolus.Capreolus_PA1_RUN1_MAXNET | PA1 | RUN1 | MAXNET | TSS | 408 | 81,632 | 56,593 | 0,382 | 0,394 | NA
13 | Capreolus.Capreolus_PA1_RUN2_GLM | PA1 | RUN2 | GLM | ROC | 515,5 | 79,441 | 58,393 | 0,744 | 0,745 | NA
14 | Capreolus.Capreolus_PA1_RUN2_GLM | PA1 | RUN2 | GLM | TSS | 514 | 79,487 | 58,171 | 0,377 | 0,376 | NA
15 | Capreolus.Capreolus_PA1_RUN2_GBM | PA1 | RUN2 | GBM | ROC | 542,5 | 78,928 | 61,886 | 0,761 | 0,751 | NA
16 | Capreolus.Capreolus_PA1_RUN2_GBM | PA1 | RUN2 | GBM | TSS | 561 | 75,711 | 65,036 | 0,407 | 0,388 | NA
17 | Capreolus.Capreolus_PA1_RUN2_GAM | PA1 | RUN2 | GAM | ROC | 478,5 | 82,517 | 57,4 | 0,755 | 0,747 | NA
18 | Capreolus.Capreolus_PA1_RUN2_GAM | PA1 | RUN2 | GAM | TSS | 476 | 82,611 | 57,036 | 0,397 | 0,385 | NA
19 | Capreolus.Capreolus_PA1_RUN2_RF | PA1 | RUN2 | RF | ROC | 353 | 100 | 99,929 | 1 | 0,723 | NA
20 | Capreolus.Capreolus_PA1_RUN2_RF | PA1 | RUN2 | RF | TSS | 347 | 100 | 99,907 | 0,999 | 0,048 | NA
21 | Capreolus.Capreolus_PA1_RUN2_MAXENT | PA1 | RUN2 | MAXENT | ROC | 422,5 | 78,322 | 62,45 | 0,766 | 0,749 | NA
22 | Capreolus.Capreolus_PA1_RUN2_MAXENT | PA1 | RUN2 | MAXENT | TSS | 422 | 78,368 | 62,264 | 0,408 | 0,379 | NA
23 | Capreolus.Capreolus_PA1_RUN2_MAXNET | PA1 | RUN2 | MAXNET | ROC | 417,5 | 80,186 | 58,85 | 0,753 | 0,751 | NA
24 | Capreolus.Capreolus_PA1_RUN2_MAXNET | PA1 | RUN2 | MAXNET | TSS | 418 | 80,186 | 58,85 | 0,389 | 0,386 | NA
25 | Capreolus.Capreolus_PA1_RUN3_GLM | PA1 | RUN3 | GLM | ROC | 518,5 | 78,974 | 59,829 | 0,751 | 0,732 | NA
26 | Capreolus.Capreolus_PA1_RUN3_GLM | PA1 | RUN3 | GLM | TSS | 513 | 79,534 | 59,2 | 0,387 | 0,354 | NA
27 | Capreolus.Capreolus_PA1_RUN3_GBM | PA1 | RUN3 | GBM | ROC | 535,5 | 80,653 | 61,921 | 0,766 | 0,736 | NA
28 | Capreolus.Capreolus_PA1_RUN3_GBM | PA1 | RUN3 | GBM | TSS | 535 | 80,746 | 61,75 | 0,426 | 0,365 | NA
29 | Capreolus.Capreolus_PA1_RUN3_GAM | PA1 | RUN3 | GAM | ROC | 489,5 | 82,331 | 58,686 | 0,76 | 0,733 | NA
30 | Capreolus.Capreolus_PA1_RUN3_GAM | PA1 | RUN3 | GAM | TSS | 488 | 82,517 | 58,493 | 0,41 | 0,368 | NA
31 | Capreolus.Capreolus_PA1_RUN3_RF | PA1 | RUN3 | RF | ROC | 368 | 100 | 99,957 | 1 | 0,716 | NA
32 | Capreolus.Capreolus_PA1_RUN3_RF | PA1 | RUN3 | RF | TSS | 365 | 100 | 99,957 | 1 | 0,047 | NA
33 | Capreolus.Capreolus_PA1_RUN3_MAXENT | PA1 | RUN3 | MAXENT | ROC | 370,5 | 83,497 | 58,414 | 0,772 | 0,735 | NA
34 | Capreolus.Capreolus_PA1_RUN3_MAXENT | PA1 | RUN3 | MAXENT | TSS | 370 | 83,543 | 58,329 | 0,419 | 0,356 | NA
35 | Capreolus.Capreolus_PA1_RUN3_MAXNET | PA1 | RUN3 | MAXNET | ROC | 405,5 | 81,352 | 58,514 | 0,758 | 0,737 | NA
36 | Capreolus.Capreolus_PA1_RUN3_MAXNET | PA1 | RUN3 | MAXNET | TSS | 406 | 81,352 | 58,514 | 0,398 | 0,357 | NA
37 | Capreolus.Capreolus_PA1_RUN4_GLM | PA1 | RUN4 | GLM | ROC | 525,5 | 78,555 | 59,757 | 0,745 | 0,745 | NA
38 | Capreolus.Capreolus_PA1_RUN4_GLM | PA1 | RUN4 | GLM | TSS | 526 | 78,555 | 59,757 | 0,383 | 0,367 | NA
39 | Capreolus.Capreolus_PA1_RUN4_GBM | PA1 | RUN4 | GBM | ROC | 555,5 | 76,643 | 65,05 | 0,76 | 0,752 | NA
40 | Capreolus.Capreolus_PA1_RUN4_GBM | PA1 | RUN4 | GBM | TSS | 555 | 76,783 | 64,779 | 0,417 | 0,389 | NA
41 | Capreolus.Capreolus_PA1_RUN4_GAM | PA1 | RUN4 | GAM | ROC | 523,5 | 78,089 | 61,993 | 0,755 | 0,746 | NA
42 | Capreolus.Capreolus_PA1_RUN4_GAM | PA1 | RUN4 | GAM | TSS | 490 | 81,492 | 58,35 | 0,399 | 0,38 | NA
43 | Capreolus.Capreolus_PA1_RUN4_RF | PA1 | RUN4 | RF | ROC | 371 | 100 | 99,964 | 1 | 0,737 | NA
44 | Capreolus.Capreolus_PA1_RUN4_RF | PA1 | RUN4 | RF | TSS | 367 | 100 | 99,957 | 1 | 0,047 | NA
45 | Capreolus.Capreolus_PA1_RUN4_MAXENT | PA1 | RUN4 | MAXENT | ROC | 424,5 | 78,182 | 62,671 | 0,765 | 0,75 | NA
46 | Capreolus.Capreolus_PA1_RUN4_MAXENT | PA1 | RUN4 | MAXENT | TSS | 405 | 80,233 | 60,293 | 0,407 | 0,379 | NA
47 | Capreolus.Capreolus_PA1_RUN4_MAXNET | PA1 | RUN4 | MAXNET | ROC | 417,5 | 80,326 | 59,014 | 0,754 | 0,749 | NA
48 | Capreolus.Capreolus_PA1_RUN4_MAXNET | PA1 | RUN4 | MAXNET | TSS | 415 | 80,513 | 58,586 | 0,392 | 0,377 | NA
49 | Capreolus.Capreolus_PA1_RUN5_GLM | PA1 | RUN5 | GLM | ROC | 512,5 | 79,86 | 58,136 | 0,745 | 0,745 | NA
50 | Capreolus.Capreolus_PA1_RUN5_GLM | PA1 | RUN5 | GLM | TSS | 501 | 81,119 | 56,729 | 0,38 | 0,377 | NA
51 | Capreolus.Capreolus_PA1_RUN5_GBM | PA1 | RUN5 | GBM | ROC | 540,5 | 79,86 | 61,386 | 0,76 | 0,751 | NA
52 | Capreolus.Capreolus_PA1_RUN5_GBM | PA1 | RUN5 | GBM | TSS | 539 | 80,093 | 60,979 | 0,411 | 0,399 | NA
53 | Capreolus.Capreolus_PA1_RUN5_GAM | PA1 | RUN5 | GAM | ROC | 471,5 | 83,73 | 56,129 | 0,752 | 0,752 | NA
54 | Capreolus.Capreolus_PA1_RUN5_GAM | PA1 | RUN5 | GAM | TSS | 475 | 83,31 | 56,471 | 0,398 | 0,389 | NA
55 | Capreolus.Capreolus_PA1_RUN5_RF | PA1 | RUN5 | RF | ROC | 361 | 100 | 99,95 | 1 | 0,734 | NA
56 | Capreolus.Capreolus_PA1_RUN5_RF | PA1 | RUN5 | RF | TSS | 361 | 100 | 99,95 | 1 | 0,049 | NA
57 | Capreolus.Capreolus_PA1_RUN5_MAXENT | PA1 | RUN5 | MAXENT | ROC | 388,5 | 82,751 | 57,879 | 0,762 | 0,754 | NA
58 | Capreolus.Capreolus_PA1_RUN5_MAXENT | PA1 | RUN5 | MAXENT | TSS | 385 | 82,844 | 57,521 | 0,404 | 0,376 | NA
59 | Capreolus.Capreolus_PA1_RUN5_MAXNET | PA1 | RUN5 | MAXNET | ROC | 398,5 | 82,751 | 55,75 | 0,75 | 0,751 | NA
60 | Capreolus.Capreolus_PA1_RUN5_MAXNET | PA1 | RUN5 | MAXNET | TSS | 396 | 83,17 | 55,271 | 0,385 | 0,384 | NA
61 | Capreolus.Capreolus_PA1_RUN6_GLM | PA1 | RUN6 | GLM | ROC | 529,5 | 78,275 | 60,379 | 0,748 | 0,736 | NA
62 | Capreolus.Capreolus_PA1_RUN6_GLM | PA1 | RUN6 | GLM | TSS | 530 | 78,275 | 60,379 | 0,386 | 0,343 | NA
63 | Capreolus.Capreolus_PA1_RUN6_GBM | PA1 | RUN6 | GBM | ROC | 525,5 | 82,051 | 59,643 | 0,762 | 0,744 | NA
64 | Capreolus.Capreolus_PA1_RUN6_GBM | PA1 | RUN6 | GBM | TSS | 524 | 82,424 | 59,257 | 0,416 | 0,376 | NA
65 | Capreolus.Capreolus_PA1_RUN6_GAM | PA1 | RUN6 | GAM | ROC | 483,5 | 82,564 | 57,771 | 0,755 | 0,746 | NA
66 | Capreolus.Capreolus_PA1_RUN6_GAM | PA1 | RUN6 | GAM | TSS | 489 | 81,958 | 58,307 | 0,403 | 0,367 | NA
67 | Capreolus.Capreolus_PA1_RUN6_RF | PA1 | RUN6 | RF | ROC | 359 | 100 | 99,964 | 1 | 0,735 | NA
68 | Capreolus.Capreolus_PA1_RUN6_RF | PA1 | RUN6 | RF | TSS | 353 | 100 | 99,957 | 1 | 0,06 | NA
69 | Capreolus.Capreolus_PA1_RUN6_MAXENT | PA1 | RUN6 | MAXENT | ROC | 407,5 | 79,814 | 61,136 | 0,765 | 0,744 | NA
70 | Capreolus.Capreolus_PA1_RUN6_MAXENT | PA1 | RUN6 | MAXENT | TSS | 409 | 79,627 | 61,286 | 0,408 | 0,368 | NA
71 | Capreolus.Capreolus_PA1_RUN6_MAXNET | PA1 | RUN6 | MAXNET | ROC | 431,5 | 78,182 | 61,236 | 0,754 | 0,742 | NA
72 | Capreolus.Capreolus_PA1_RUN6_MAXNET | PA1 | RUN6 | MAXNET | TSS | 436 | 77,576 | 61,736 | 0,394 | 0,365 | NA
73 | Capreolus.Capreolus_PA1_RUN7_GLM | PA1 | RUN7 | GLM | ROC | 507,5 | 80,979 | 57,657 | 0,751 | 0,733 | NA
74 | Capreolus.Capreolus_PA1_RUN7_GLM | PA1 | RUN7 | GLM | TSS | 520 | 79,674 | 58,843 | 0,386 | 0,36 | NA
75 | Capreolus.Capreolus_PA1_RUN7_GBM | PA1 | RUN7 | GBM | ROC | 530,5 | 80,886 | 60,607 | 0,764 | 0,741 | NA
76 | Capreolus.Capreolus_PA1_RUN7_GBM | PA1 | RUN7 | GBM | TSS | 528 | 81,445 | 60 | 0,415 | 0,386 | NA
77 | Capreolus.Capreolus_PA1_RUN7_GAM | PA1 | RUN7 | GAM | ROC | 523,5 | 78,741 | 62,35 | 0,759 | 0,737 | NA
78 | Capreolus.Capreolus_PA1_RUN7_GAM | PA1 | RUN7 | GAM | TSS | 528 | 77,855 | 62,779 | 0,407 | 0,357 | NA
79 | Capreolus.Capreolus_PA1_RUN7_RF | PA1 | RUN7 | RF | ROC | 363 | 99,907 | 99,929 | 1 | 0,717 | NA
80 | Capreolus.Capreolus_PA1_RUN7_RF | PA1 | RUN7 | RF | TSS | 364 | 99,907 | 99,929 | 0,998 | 0,041 | NA
81 | Capreolus.Capreolus_PA1_RUN7_MAXENT | PA1 | RUN7 | MAXENT | ROC | 420,5 | 77,762 | 63,343 | 0,769 | 0,737 | NA
82 | Capreolus.Capreolus_PA1_RUN7_MAXENT | PA1 | RUN7 | MAXENT | TSS | 422 | 77,622 | 63,436 | 0,411 | 0,35 | NA
83 | Capreolus.Capreolus_PA1_RUN7_MAXNET | PA1 | RUN7 | MAXNET | ROC | 415,5 | 80 | 59,414 | 0,756 | 0,738 | NA
84 | Capreolus.Capreolus_PA1_RUN7_MAXNET | PA1 | RUN7 | MAXNET | TSS | 418 | 79,72 | 59,65 | 0,393 | 0,366 | NA
85 | Capreolus.Capreolus_PA1_RUN8_GLM | PA1 | RUN8 | GLM | ROC | 521,5 | 78,974 | 59,807 | 0,751 | 0,73 | NA
86 | Capreolus.Capreolus_PA1_RUN8_GLM | PA1 | RUN8 | GLM | TSS | 524 | 78,741 | 59,971 | 0,388 | 0,348 | NA
87 | Capreolus.Capreolus_PA1_RUN8_GBM | PA1 | RUN8 | GBM | ROC | 542,5 | 78,881 | 63,536 | 0,767 | 0,736 | NA
88 | Capreolus.Capreolus_PA1_RUN8_GBM | PA1 | RUN8 | GBM | TSS | 543 | 78,881 | 63,536 | 0,424 | 0,369 | NA
89 | Capreolus.Capreolus_PA1_RUN8_GAM | PA1 | RUN8 | GAM | ROC | 514,5 | 79,254 | 61,407 | 0,759 | 0,739 | NA
90 | Capreolus.Capreolus_PA1_RUN8_GAM | PA1 | RUN8 | GAM | TSS | 501 | 80,653 | 59,836 | 0,405 | 0,364 | NA
91 | Capreolus.Capreolus_PA1_RUN8_RF | PA1 | RUN8 | RF | ROC | 360 | 100 | 99,921 | 1 | 0,725 | NA
92 | Capreolus.Capreolus_PA1_RUN8_RF | PA1 | RUN8 | RF | TSS | 360 | 100 | 99,921 | 0,999 | 0,046 | NA
93 | Capreolus.Capreolus_PA1_RUN8_MAXENT | PA1 | RUN8 | MAXENT | ROC | 427,5 | 76,41 | 64,693 | 0,769 | 0,736 | NA
94 | Capreolus.Capreolus_PA1_RUN8_MAXENT | PA1 | RUN8 | MAXENT | TSS | 426 | 76,643 | 64,386 | 0,411 | 0,361 | NA
95 | Capreolus.Capreolus_PA1_RUN8_MAXNET | PA1 | RUN8 | MAXNET | ROC | 434,5 | 77,249 | 62,421 | 0,758 | 0,733 | NA
96 | Capreolus.Capreolus_PA1_RUN8_MAXNET | PA1 | RUN8 | MAXNET | TSS | 432 | 77,529 | 61,986 | 0,396 | 0,353 | NA
97 | Capreolus.Capreolus_PA1_RUN9_GLM | PA1 | RUN9 | GLM | ROC | 515,5 | 79,907 | 57,914 | 0,747 | 0,742 | NA
98 | Capreolus.Capreolus_PA1_RUN9_GLM | PA1 | RUN9 | GLM | TSS | 515 | 79,953 | 57,743 | 0,378 | 0,378 | NA
99 | Capreolus.Capreolus_PA1_RUN9_GBM | PA1 | RUN9 | GBM | ROC | 555,5 | 76,503 | 64,436 | 0,764 | 0,744 | NA
100 | Capreolus.Capreolus_PA1_RUN9_GBM | PA1 | RUN9 | GBM | TSS | 565 | 74,545 | 66,229 | 0,409 | 0,383 | NA
101 | Capreolus.Capreolus_PA1_RUN9_GAM | PA1 | RUN9 | GAM | ROC | 509,5 | 80,233 | 60 | 0,757 | 0,741 | NA
102 | Capreolus.Capreolus_PA1_RUN9_GAM | PA1 | RUN9 | GAM | TSS | 503 | 80,746 | 59,2 | 0,4 | 0,375 | NA
103 | Capreolus.Capreolus_PA1_RUN9_RF | PA1 | RUN9 | RF | ROC | 359 | 99,953 | 99,929 | 1 | 0,727 | NA
104 | Capreolus.Capreolus_PA1_RUN9_RF | PA1 | RUN9 | RF | TSS | 354 | 99,953 | 99,914 | 0,999 | 0,058 | NA
105 | Capreolus.Capreolus_PA1_RUN9_MAXENT | PA1 | RUN9 | MAXENT | ROC | 376,5 | 82,751 | 58,407 | 0,77 | 0,742 | NA
106 | Capreolus.Capreolus_PA1_RUN9_MAXENT | PA1 | RUN9 | MAXENT | TSS | 376 | 82,797 | 58,343 | 0,412 | 0,374 | NA
107 | Capreolus.Capreolus_PA1_RUN9_MAXNET | PA1 | RUN9 | MAXNET | ROC | 410,5 | 80,746 | 58,307 | 0,756 | 0,744 | NA
108 | Capreolus.Capreolus_PA1_RUN9_MAXNET | PA1 | RUN9 | MAXNET | TSS | 433 | 77,669 | 61,121 | 0,389 | 0,387 | NA
109 | Capreolus.Capreolus_PA1_RUN10_GLM | PA1 | RUN10 | GLM | ROC | 555,5 | 74,732 | 63,186 | 0,749 | 0,734 | NA
110 | Capreolus.Capreolus_PA1_RUN10_GLM | PA1 | RUN10 | GLM | TSS | 509 | 79,86 | 57,914 | 0,379 | 0,362 | NA
111 | Capreolus.Capreolus_PA1_RUN10_GBM | PA1 | RUN10 | GBM | ROC | 544,5 | 78,648 | 62,814 | 0,764 | 0,739 | NA
112 | Capreolus.Capreolus_PA1_RUN10_GBM | PA1 | RUN10 | GBM | TSS | 537 | 79,674 | 61,643 | 0,414 | 0,383 | NA
113 | Capreolus.Capreolus_PA1_RUN10_GAM | PA1 | RUN10 | GAM | ROC | 504,5 | 79,953 | 60,407 | 0,757 | 0,74 | NA
114 | Capreolus.Capreolus_PA1_RUN10_GAM | PA1 | RUN10 | GAM | TSS | 502 | 80,047 | 60,15 | 0,401 | 0,374 | NA
115 | Capreolus.Capreolus_PA1_RUN10_RF | PA1 | RUN10 | RF | ROC | 383 | 100 | 99,957 | 1 | 0,721 | NA
116 | Capreolus.Capreolus_PA1_RUN10_RF | PA1 | RUN10 | RF | TSS | 383 | 100 | 99,957 | 1 | 0,028 | NA
117 | Capreolus.Capreolus_PA1_RUN10_MAXENT | PA1 | RUN10 | MAXENT | ROC | 404,5 | 79,953 | 61,25 | 0,767 | 0,738 | NA
118 | Capreolus.Capreolus_PA1_RUN10_MAXENT | PA1 | RUN10 | MAXENT | TSS | 405 | 79,953 | 61,25 | 0,412 | 0,378 | NA
119 | Capreolus.Capreolus_PA1_RUN10_MAXNET | PA1 | RUN10 | MAXNET | ROC | 430,5 | 77,902 | 61,243 | 0,755 | 0,739 | NA
120 | Capreolus.Capreolus_PA1_RUN10_MAXNET | PA1 | RUN10 | MAXNET | TSS | 449 | 75,385 | 63,643 | 0,391 | 0,369 | NA

</body>

</html>

```r
calib.lines <- get_calib_lines(Capreolus_single_models, PA = "allData", run = "allRun")
biomod2::plot(Capreolus_BIOMOD_data, calib.lines = calib.lines,
              plot.type = "raster", plot.eval = T, do.plot = T)
``` 

![Rplot01](https://github.com/biomodhub/biomod2/assets/121942966/6ab67b10-f282-4473-8780-352b81dc3b8e)

I really hope you can help me and give me some advice with these information, that would really help me!!!
Thank you so much in advance for all your time, I wish you a great day!

_Originally posted by @LorenzoBernicchi in https://github.com/biomodhub/biomod2/issues/293#issuecomment-1643585318_

**Comment by MayaGueguen** [comment]
Hello Lorenzo,

I opened a new issue as this is not related with your previous GAM error and is a brand new set of modeling.
Please do not hesitate to separate issues when topics differ 🙏 
It is easier for us (and for interested users) to follow :) 

Thank you for all your detailed informations.

I'm a bit surprised by the differences between your `Capreolus_BIOMOD_data` summary, indicating 3064 presences and 20000 pseudo-absences, and your corresponding plot, obtained with lines : 
```
calib.lines <- get_calib_lines(Capreolus_single_models, PA = "allData", run = "allRun")
biomod2::plot(Capreolus_BIOMOD_data, calib.lines = calib.lines,
              plot.type = "raster", plot.eval = T, do.plot = T)
```
It looks like on this plot that you change very few presences and pseudo-absences points, but it might be the plot function that is not rendering well ?
Could you check if running the [BIOMOD_FormatingData](https://biomodhub.github.io/biomod2/reference/BIOMOD_FormatingData.html) function with `filter.raster = TRUE` you get the same number of points within you formated data summary ?

Anyhow, this seems to be a random forest relative "issue" as you discussed before with Rémi (see https://github.com/biomodhub/biomod2/issues/261#issuecomment-1598798884)
Did you try what he suggested, changing the `nodesize` options before running your model ?

> `
> myBIOMODoption <- BIOMOD_ModelingOptions(RF = list(nodesize = 10))
> `
> Increasing nodesize (from 5 to 10) should lead to smaller tree, less prone to overfit. Alternatively you can also try setting a value to maxnodes that should also lead to smaller tree or forest which might as well limit overfit.

Maya

**Comment by LorenzoBernicchi** [comment]
Dear @MayaGueguen,
Thank you so much for opening a new issue, honestly I didn't think about it. 

I checked my `BIOMOD_FormatingData` using `filter.raster = TRUE` and actually I get the same number of points, 3064.

Regarding the `randomForest` I tried to change these options:
```r
myBIOMODoption <- BIOMOD_ModelingOptions(RF = list(do.classif = F,
                                                   ntree = 2500,
                                                   sampsize = c(2000, 2000),
                                                   mtry = 4,
                                                   nodesize = 10))
``` 
Using these options I get slightly different results, hereafter you can see the scores of the single algorithms:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Loren/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Loren/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



  | full.name | PA | run | algo | metric.eval | cutoff | sensitivity | specificity | calibration | validation | evaluation
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | Capreolus.Capreolus_PA1_RUN1_GLM | PA1 | RUN1 | GLM | ROC | 518,5 | 79,301 | 58,821 | 0,745 | 0,745 | NA
2 | Capreolus.Capreolus_PA1_RUN1_GLM | PA1 | RUN1 | GLM | TSS | 516 | 79,441 | 58,586 | 0,381 | 0,379 | NA
3 | Capreolus.Capreolus_PA1_RUN1_GBM | PA1 | RUN1 | GBM | ROC | 550,5 | 78,135 | 62,8 | 0,76 | 0,751 | NA
4 | Capreolus.Capreolus_PA1_RUN1_GBM | PA1 | RUN1 | GBM | TSS | 551 | 78,135 | 62,8 | 0,409 | 0,395 | NA
5 | Capreolus.Capreolus_PA1_RUN1_GAM | PA1 | RUN1 | GAM | ROC | 502,5 | 80,979 | 58,843 | 0,753 | 0,749 | NA
6 | Capreolus.Capreolus_PA1_RUN1_GAM | PA1 | RUN1 | GAM | TSS | 503 | 80,979 | 58,843 | 0,397 | 0,399 | NA
7 | Capreolus.Capreolus_PA1_RUN1_RF | PA1 | RUN1 | RF | ROC | 618,5 | 87,646 | 87,843 | 0,958 | 0,753 | NA
8 | Capreolus.Capreolus_PA1_RUN1_RF | PA1 | RUN1 | RF | TSS | 628 | 86,527 | 88,879 | 0,755 | 0,218 | NA
9 | Capreolus.Capreolus_PA1_RUN1_MAXENT | PA1 | RUN1 | MAXENT | ROC | 438,5 | 77,156 | 63,107 | 0,763 | 0,75 | NA
10 | Capreolus.Capreolus_PA1_RUN1_MAXENT | PA1 | RUN1 | MAXENT | TSS | 448 | 75,664 | 64,507 | 0,402 | 0,395 | NA
11 | Capreolus.Capreolus_PA1_RUN1_MAXNET | PA1 | RUN1 | MAXNET | ROC | 433,5 | 78,415 | 60,4 | 0,75 | 0,75 | NA
12 | Capreolus.Capreolus_PA1_RUN1_MAXNET | PA1 | RUN1 | MAXNET | TSS | 430 | 78,695 | 59,964 | 0,387 | 0,388 | NA
13 | Capreolus.Capreolus_PA1_RUN2_GLM | PA1 | RUN2 | GLM | ROC | 507,5 | 79,953 | 57,593 | 0,743 | 0,749 | NA
14 | Capreolus.Capreolus_PA1_RUN2_GLM | PA1 | RUN2 | GLM | TSS | 505 | 80,093 | 57,207 | 0,375 | 0,386 | NA
15 | Capreolus.Capreolus_PA1_RUN2_GBM | PA1 | RUN2 | GBM | ROC | 529,5 | 80,793 | 60,364 | 0,759 | 0,751 | NA
16 | Capreolus.Capreolus_PA1_RUN2_GBM | PA1 | RUN2 | GBM | TSS | 531 | 80,606 | 60,55 | 0,41 | 0,402 | NA
17 | Capreolus.Capreolus_PA1_RUN2_GAM | PA1 | RUN2 | GAM | ROC | 497,5 | 80,746 | 59,3 | 0,752 | 0,751 | NA
18 | Capreolus.Capreolus_PA1_RUN2_GAM | PA1 | RUN2 | GAM | TSS | 504 | 80,093 | 59,886 | 0,4 | 0,402 | NA
19 | Capreolus.Capreolus_PA1_RUN2_RF | PA1 | RUN2 | RF | ROC | 631,5 | 86,527 | 89,614 | 0,96 | 0,752 | NA
20 | Capreolus.Capreolus_PA1_RUN2_RF | PA1 | RUN2 | RF | TSS | 635 | 86,014 | 90,036 | 0,76 | 0,205 | NA
21 | Capreolus.Capreolus_PA1_RUN2_MAXENT | PA1 | RUN2 | MAXENT | ROC | 432,5 | 76,876 | 63,464 | 0,761 | 0,753 | NA
22 | Capreolus.Capreolus_PA1_RUN2_MAXENT | PA1 | RUN2 | MAXENT | TSS | 418 | 78,881 | 61,279 | 0,403 | 0,4 | NA
23 | Capreolus.Capreolus_PA1_RUN2_MAXNET | PA1 | RUN2 | MAXNET | ROC | 410,5 | 81,166 | 58,029 | 0,75 | 0,752 | NA
24 | Capreolus.Capreolus_PA1_RUN2_MAXNET | PA1 | RUN2 | MAXNET | TSS | 409 | 81,212 | 57,771 | 0,391 | 0,394 | NA
25 | Capreolus.Capreolus_PA1_RUN3_GLM | PA1 | RUN3 | GLM | ROC | 530,5 | 77,902 | 60,471 | 0,747 | 0,739 | NA
26 | Capreolus.Capreolus_PA1_RUN3_GLM | PA1 | RUN3 | GLM | TSS | 526 | 78,322 | 59,957 | 0,382 | 0,363 | NA
27 | Capreolus.Capreolus_PA1_RUN3_GBM | PA1 | RUN3 | GBM | ROC | 547,5 | 77,716 | 64,371 | 0,764 | 0,745 | NA
28 | Capreolus.Capreolus_PA1_RUN3_GBM | PA1 | RUN3 | GBM | TSS | 546 | 77,762 | 64,1 | 0,42 | 0,374 | NA
29 | Capreolus.Capreolus_PA1_RUN3_GAM | PA1 | RUN3 | GAM | ROC | 507,5 | 80,14 | 60,707 | 0,756 | 0,741 | NA
30 | Capreolus.Capreolus_PA1_RUN3_GAM | PA1 | RUN3 | GAM | TSS | 506 | 80,326 | 60,486 | 0,407 | 0,367 | NA
31 | Capreolus.Capreolus_PA1_RUN3_RF | PA1 | RUN3 | RF | ROC | 641,5 | 84,103 | 90,564 | 0,957 | 0,743 | NA
32 | Capreolus.Capreolus_PA1_RUN3_RF | PA1 | RUN3 | RF | TSS | 635 | 84,802 | 89,671 | 0,745 | 0,165 | NA
33 | Capreolus.Capreolus_PA1_RUN3_MAXENT | PA1 | RUN3 | MAXENT | ROC | 432,5 | 76,83 | 64,907 | 0,767 | 0,742 | NA
34 | Capreolus.Capreolus_PA1_RUN3_MAXENT | PA1 | RUN3 | MAXENT | TSS | 433 | 76,83 | 64,907 | 0,416 | 0,348 | NA
35 | Capreolus.Capreolus_PA1_RUN3_MAXNET | PA1 | RUN3 | MAXNET | ROC | 430,5 | 78,695 | 61,029 | 0,754 | 0,744 | NA
36 | Capreolus.Capreolus_PA1_RUN3_MAXNET | PA1 | RUN3 | MAXNET | TSS | 438 | 77,622 | 61,907 | 0,396 | 0,36 | NA
37 | Capreolus.Capreolus_PA1_RUN4_GLM | PA1 | RUN4 | GLM | ROC | 504,5 | 81,352 | 56,829 | 0,746 | 0,743 | NA
38 | Capreolus.Capreolus_PA1_RUN4_GLM | PA1 | RUN4 | GLM | TSS | 508 | 80,932 | 57,129 | 0,38 | 0,369 | NA
39 | Capreolus.Capreolus_PA1_RUN4_GBM | PA1 | RUN4 | GBM | ROC | 536,5 | 81,212 | 60,079 | 0,761 | 0,744 | NA
40 | Capreolus.Capreolus_PA1_RUN4_GBM | PA1 | RUN4 | GBM | TSS | 533 | 81,725 | 59,464 | 0,412 | 0,382 | NA
41 | Capreolus.Capreolus_PA1_RUN4_GAM | PA1 | RUN4 | GAM | ROC | 525,5 | 77,855 | 61,314 | 0,753 | 0,749 | NA
42 | Capreolus.Capreolus_PA1_RUN4_GAM | PA1 | RUN4 | GAM | TSS | 466 | 84,382 | 54,643 | 0,39 | 0,384 | NA
43 | Capreolus.Capreolus_PA1_RUN4_RF | PA1 | RUN4 | RF | ROC | 645,5 | 84,615 | 91,614 | 0,96 | 0,744 | NA
44 | Capreolus.Capreolus_PA1_RUN4_RF | PA1 | RUN4 | RF | TSS | 648 | 84,336 | 91,807 | 0,761 | 0,184 | NA
45 | Capreolus.Capreolus_PA1_RUN4_MAXENT | PA1 | RUN4 | MAXENT | ROC | 366,5 | 83,916 | 56,593 | 0,764 | 0,747 | NA
46 | Capreolus.Capreolus_PA1_RUN4_MAXENT | PA1 | RUN4 | MAXENT | TSS | 366 | 83,916 | 56,493 | 0,405 | 0,37 | NA
47 | Capreolus.Capreolus_PA1_RUN4_MAXNET | PA1 | RUN4 | MAXNET | ROC | 422,5 | 79,254 | 59,579 | 0,752 | 0,748 | NA
48 | Capreolus.Capreolus_PA1_RUN4_MAXNET | PA1 | RUN4 | MAXNET | TSS | 417 | 79,814 | 58,771 | 0,387 | 0,373 | NA
49 | Capreolus.Capreolus_PA1_RUN5_GLM | PA1 | RUN5 | GLM | ROC | 533,5 | 76,876 | 60,071 | 0,74 | 0,757 | NA
50 | Capreolus.Capreolus_PA1_RUN5_GLM | PA1 | RUN5 | GLM | TSS | 538 | 76,41 | 60,464 | 0,369 | 0,391 | NA
51 | Capreolus.Capreolus_PA1_RUN5_GBM | PA1 | RUN5 | GBM | ROC | 516,5 | 82,191 | 58,057 | 0,757 | 0,761 | NA
52 | Capreolus.Capreolus_PA1_RUN5_GBM | PA1 | RUN5 | GBM | TSS | 514 | 82,424 | 57,707 | 0,402 | 0,41 | NA
53 | Capreolus.Capreolus_PA1_RUN5_GAM | PA1 | RUN5 | GAM | ROC | 510,5 | 79,394 | 60,264 | 0,749 | 0,759 | NA
54 | Capreolus.Capreolus_PA1_RUN5_GAM | PA1 | RUN5 | GAM | TSS | 504 | 80,093 | 59,336 | 0,395 | 0,399 | NA
55 | Capreolus.Capreolus_PA1_RUN5_RF | PA1 | RUN5 | RF | ROC | 633,5 | 86,62 | 90,321 | 0,96 | 0,752 | NA
56 | Capreolus.Capreolus_PA1_RUN5_RF | PA1 | RUN5 | RF | TSS | 633 | 86,667 | 90,193 | 0,769 | 0,196 | NA
57 | Capreolus.Capreolus_PA1_RUN5_MAXENT | PA1 | RUN5 | MAXENT | ROC | 379,5 | 82,751 | 57,136 | 0,76 | 0,761 | NA
58 | Capreolus.Capreolus_PA1_RUN5_MAXENT | PA1 | RUN5 | MAXENT | TSS | 378 | 82,844 | 56,85 | 0,397 | 0,405 | NA
59 | Capreolus.Capreolus_PA1_RUN5_MAXNET | PA1 | RUN5 | MAXNET | ROC | 440,5 | 77,156 | 60,807 | 0,746 | 0,763 | NA
60 | Capreolus.Capreolus_PA1_RUN5_MAXNET | PA1 | RUN5 | MAXNET | TSS | 408 | 81,772 | 56,093 | 0,378 | 0,406 | NA
61 | Capreolus.Capreolus_PA1_RUN6_GLM | PA1 | RUN6 | GLM | ROC | 504,5 | 81,818 | 56,993 | 0,747 | 0,74 | NA
62 | Capreolus.Capreolus_PA1_RUN6_GLM | PA1 | RUN6 | GLM | TSS | 504 | 81,865 | 56,864 | 0,388 | 0,36 | NA
63 | Capreolus.Capreolus_PA1_RUN6_GBM | PA1 | RUN6 | GBM | ROC | 535,5 | 81,259 | 60,864 | 0,764 | 0,743 | NA
64 | Capreolus.Capreolus_PA1_RUN6_GBM | PA1 | RUN6 | GBM | TSS | 546 | 79,301 | 62,564 | 0,42 | 0,38 | NA
65 | Capreolus.Capreolus_PA1_RUN6_GAM | PA1 | RUN6 | GAM | ROC | 489,5 | 81,818 | 58,836 | 0,757 | 0,739 | NA
66 | Capreolus.Capreolus_PA1_RUN6_GAM | PA1 | RUN6 | GAM | TSS | 478 | 83,124 | 57,393 | 0,406 | 0,382 | NA
67 | Capreolus.Capreolus_PA1_RUN6_RF | PA1 | RUN6 | RF | ROC | 640,5 | 85,035 | 90,543 | 0,958 | 0,743 | NA
68 | Capreolus.Capreolus_PA1_RUN6_RF | PA1 | RUN6 | RF | TSS | 645 | 84,522 | 91,043 | 0,755 | 0,174 | NA
69 | Capreolus.Capreolus_PA1_RUN6_MAXENT | PA1 | RUN6 | MAXENT | ROC | 434,5 | 77,296 | 64,414 | 0,77 | 0,737 | NA
70 | Capreolus.Capreolus_PA1_RUN6_MAXENT | PA1 | RUN6 | MAXENT | TSS | 437 | 77,063 | 64,629 | 0,417 | 0,369 | NA
71 | Capreolus.Capreolus_PA1_RUN6_MAXNET | PA1 | RUN6 | MAXNET | ROC | 398,5 | 83,263 | 56,286 | 0,755 | 0,742 | NA
72 | Capreolus.Capreolus_PA1_RUN6_MAXNET | PA1 | RUN6 | MAXNET | TSS | 397 | 83,45 | 56,057 | 0,395 | 0,368 | NA
73 | Capreolus.Capreolus_PA1_RUN7_GLM | PA1 | RUN7 | GLM | ROC | 532,5 | 77,762 | 60,607 | 0,748 | 0,74 | NA
74 | Capreolus.Capreolus_PA1_RUN7_GLM | PA1 | RUN7 | GLM | TSS | 530 | 77,995 | 60,279 | 0,383 | 0,37 | NA
75 | Capreolus.Capreolus_PA1_RUN7_GBM | PA1 | RUN7 | GBM | ROC | 543,5 | 79,953 | 62,057 | 0,763 | 0,747 | NA
76 | Capreolus.Capreolus_PA1_RUN7_GBM | PA1 | RUN7 | GBM | TSS | 540 | 80,653 | 61,157 | 0,419 | 0,383 | NA
77 | Capreolus.Capreolus_PA1_RUN7_GAM | PA1 | RUN7 | GAM | ROC | 461,5 | 85,361 | 55,164 | 0,756 | 0,744 | NA
78 | Capreolus.Capreolus_PA1_RUN7_GAM | PA1 | RUN7 | GAM | TSS | 459 | 85,548 | 54,936 | 0,405 | 0,375 | NA
79 | Capreolus.Capreolus_PA1_RUN7_RF | PA1 | RUN7 | RF | ROC | 649,5 | 84,942 | 91,564 | 0,959 | 0,748 | NA
80 | Capreolus.Capreolus_PA1_RUN7_RF | PA1 | RUN7 | RF | TSS | 645 | 85,408 | 91 | 0,764 | 0,185 | NA
81 | Capreolus.Capreolus_PA1_RUN7_MAXENT | PA1 | RUN7 | MAXENT | ROC | 436,5 | 76,503 | 64,793 | 0,767 | 0,747 | NA
82 | Capreolus.Capreolus_PA1_RUN7_MAXENT | PA1 | RUN7 | MAXENT | TSS | 438 | 76,27 | 64,936 | 0,412 | 0,369 | NA
83 | Capreolus.Capreolus_PA1_RUN7_MAXNET | PA1 | RUN7 | MAXNET | ROC | 397,5 | 82,145 | 56,764 | 0,754 | 0,747 | NA
84 | Capreolus.Capreolus_PA1_RUN7_MAXNET | PA1 | RUN7 | MAXNET | TSS | 439 | 76,97 | 61,829 | 0,388 | 0,374 | NA
85 | Capreolus.Capreolus_PA1_RUN8_GLM | PA1 | RUN8 | GLM | ROC | 503,5 | 80,047 | 57,429 | 0,741 | 0,751 | NA
86 | Capreolus.Capreolus_PA1_RUN8_GLM | PA1 | RUN8 | GLM | TSS | 503 | 80,047 | 57,271 | 0,375 | 0,383 | NA
87 | Capreolus.Capreolus_PA1_RUN8_GBM | PA1 | RUN8 | GBM | ROC | 546,5 | 78,089 | 62,871 | 0,758 | 0,757 | NA
88 | Capreolus.Capreolus_PA1_RUN8_GBM | PA1 | RUN8 | GBM | TSS | 551 | 77,016 | 63,829 | 0,408 | 0,392 | NA
89 | Capreolus.Capreolus_PA1_RUN8_GAM | PA1 | RUN8 | GAM | ROC | 512,5 | 79,161 | 60,7 | 0,753 | 0,748 | NA
90 | Capreolus.Capreolus_PA1_RUN8_GAM | PA1 | RUN8 | GAM | TSS | 513 | 79,161 | 60,7 | 0,397 | 0,388 | NA
91 | Capreolus.Capreolus_PA1_RUN8_RF | PA1 | RUN8 | RF | ROC | 638,5 | 84,988 | 90,814 | 0,958 | 0,756 | NA
92 | Capreolus.Capreolus_PA1_RUN8_RF | PA1 | RUN8 | RF | TSS | 633 | 85,455 | 90,143 | 0,756 | 0,189 | NA
93 | Capreolus.Capreolus_PA1_RUN8_MAXENT | PA1 | RUN8 | MAXENT | ROC | 454,5 | 74,918 | 66,121 | 0,764 | 0,756 | NA
94 | Capreolus.Capreolus_PA1_RUN8_MAXENT | PA1 | RUN8 | MAXENT | TSS | 456 | 74,732 | 66,286 | 0,41 | 0,381 | NA
95 | Capreolus.Capreolus_PA1_RUN8_MAXNET | PA1 | RUN8 | MAXNET | ROC | 458,5 | 75,431 | 63,557 | 0,75 | 0,755 | NA
96 | Capreolus.Capreolus_PA1_RUN8_MAXNET | PA1 | RUN8 | MAXNET | TSS | 411 | 81,445 | 57,364 | 0,389 | 0,397 | NA
97 | Capreolus.Capreolus_PA1_RUN9_GLM | PA1 | RUN9 | GLM | ROC | 533,5 | 77,576 | 59,971 | 0,744 | 0,748 | NA
98 | Capreolus.Capreolus_PA1_RUN9_GLM | PA1 | RUN9 | GLM | TSS | 502 | 81,166 | 56,307 | 0,375 | 0,384 | NA
99 | Capreolus.Capreolus_PA1_RUN9_GBM | PA1 | RUN9 | GBM | ROC | 518,5 | 81,865 | 58,671 | 0,76 | 0,755 | NA
100 | Capreolus.Capreolus_PA1_RUN9_GBM | PA1 | RUN9 | GBM | TSS | 525 | 80,793 | 59,621 | 0,405 | 0,401 | NA
101 | Capreolus.Capreolus_PA1_RUN9_GAM | PA1 | RUN9 | GAM | ROC | 515,5 | 78,601 | 61,371 | 0,752 | 0,752 | NA
102 | Capreolus.Capreolus_PA1_RUN9_GAM | PA1 | RUN9 | GAM | TSS | 515 | 78,648 | 61,264 | 0,4 | 0,373 | NA
103 | Capreolus.Capreolus_PA1_RUN9_RF | PA1 | RUN9 | RF | ROC | 628,5 | 86,2 | 89,371 | 0,958 | 0,748 | NA
104 | Capreolus.Capreolus_PA1_RUN9_RF | PA1 | RUN9 | RF | TSS | 630 | 86,061 | 89,457 | 0,755 | 0,186 | NA
105 | Capreolus.Capreolus_PA1_RUN9_MAXENT | PA1 | RUN9 | MAXENT | ROC | 382,5 | 81,119 | 58,964 | 0,763 | 0,752 | NA
106 | Capreolus.Capreolus_PA1_RUN9_MAXENT | PA1 | RUN9 | MAXENT | TSS | 384 | 80,932 | 59,071 | 0,401 | 0,388 | NA
107 | Capreolus.Capreolus_PA1_RUN9_MAXNET | PA1 | RUN9 | MAXNET | ROC | 435,5 | 76,923 | 61,707 | 0,75 | 0,752 | NA
108 | Capreolus.Capreolus_PA1_RUN9_MAXNET | PA1 | RUN9 | MAXNET | TSS | 434 | 77,11 | 61,407 | 0,386 | 0,378 | NA
109 | Capreolus.Capreolus_PA1_RUN10_GLM | PA1 | RUN10 | GLM | ROC | 512,5 | 80,28 | 58,043 | 0,747 | 0,742 | NA
110 | Capreolus.Capreolus_PA1_RUN10_GLM | PA1 | RUN10 | GLM | TSS | 511 | 80,373 | 57,764 | 0,382 | 0,363 | NA
111 | Capreolus.Capreolus_PA1_RUN10_GBM | PA1 | RUN10 | GBM | ROC | 545,5 | 80,559 | 61,007 | 0,762 | 0,748 | NA
112 | Capreolus.Capreolus_PA1_RUN10_GBM | PA1 | RUN10 | GBM | TSS | 542 | 80,979 | 60,421 | 0,415 | 0,371 | NA
113 | Capreolus.Capreolus_PA1_RUN10_GAM | PA1 | RUN10 | GAM | ROC | 485,5 | 82,191 | 57,786 | 0,756 | 0,741 | NA
114 | Capreolus.Capreolus_PA1_RUN10_GAM | PA1 | RUN10 | GAM | TSS | 486 | 82,191 | 57,786 | 0,399 | 0,384 | NA
115 | Capreolus.Capreolus_PA1_RUN10_RF | PA1 | RUN10 | RF | ROC | 627,5 | 87,552 | 88,9 | 0,958 | 0,745 | NA
116 | Capreolus.Capreolus_PA1_RUN10_RF | PA1 | RUN10 | RF | TSS | 627 | 87,646 | 88,8 | 0,765 | 0,221 | NA
117 | Capreolus.Capreolus_PA1_RUN10_MAXENT | PA1 | RUN10 | MAXENT | ROC | 442,5 | 77,016 | 64,6 | 0,77 | 0,742 | NA
118 | Capreolus.Capreolus_PA1_RUN10_MAXENT | PA1 | RUN10 | MAXENT | TSS | 461 | 74,685 | 66,757 | 0,414 | 0,365 | NA
119 | Capreolus.Capreolus_PA1_RUN10_MAXNET | PA1 | RUN10 | MAXNET | ROC | 410,5 | 81,585 | 57,907 | 0,754 | 0,746 | NA
120 | Capreolus.Capreolus_PA1_RUN10_MAXNET | PA1 | RUN10 | MAXNET | TSS | 405 | 82,191 | 57,079 | 0,394 | 0,372 | NA



</body>

</html>

As you can see the results are now a bit better than before, even though the `randomForest` still has very high scores of `AUC` and very low scores of `TSS` for the validation dataset.

**Comment by rpatin** [comment]
Hello Lorenzo,
Thank you for the update.
Good to see that adjusting `nodesize` does improve performance on the validation dataset, although it is still limited. You can try:
- to increase `nodesize` further
- to set `maxnodes` to e.g. 3, 5 or 10 (not sure what would be a reference value for `maxnodes`) and see whether it does something or not.

It would be good to also plot the response curves so that you can:
- compare RF responses across the parameters you set (`maxnodes`, `nodesize`)
- compare RF responses with other models responses to understand why are performance worse for RF

Concerning the rather bad performance of other models, I would also be curious to have more information about the dataset. The plot indeed suggests that data are pretty clumped together. But it is rather curious that we do not see the 20000 pseudo-absences points. Maybe you can try again the plot with `plot.type = "points"` ?

Best, 
Rémi

**Comment by LorenzoBernicchi** [comment]
Hello @rpatin,

Thank you so much for your answer and all your advices, I really appreciate it!

In these days I will try to adjust further the `nodesize` and `maxnodes` parameters of `randomForest`, then I will show you the scores to see if they get better.

Regarding the `response curves`: hereafter I show you the plot of the six different algorithms, but to be honest I cannot interprete them to understand what could be the potential problem.

- `GLM` response curves 
![image](https://github.com/biomodhub/biomod2/assets/121942966/a9086d2a-8f09-4fa6-9a3d-d020bd07bebc)
- `GBM` response curves
![image](https://github.com/biomodhub/biomod2/assets/121942966/0b7f1124-ddb4-4cbc-9557-cda6ba17c335)
- `GAM` response curves
![image](https://github.com/biomodhub/biomod2/assets/121942966/35c2fb0e-fe53-41f8-8f50-85fe2be89580)
- `RF` response curves
![image](https://github.com/biomodhub/biomod2/assets/121942966/0be2a8dd-afd2-439b-a77c-c2654572ea57)
About `randomForest` response curves, would you suggest me to try looking at them for the model run with the default options?
- `MAXENT` response curves
![image](https://github.com/biomodhub/biomod2/assets/121942966/bee496b3-10e4-4d95-bbb9-1c15b128834f)
- `MAXNET` response curves
![image](https://github.com/biomodhub/biomod2/assets/121942966/3715d71d-9a98-4c8a-9aa1-23eba9413279)

Regarding the dataset: I downloaded my presence points from GBIF and I properly cleaned them (with the `coordinateCleaner` package and by thinning them to 20 Km, going from more than 1 million of points to the 3000 I am using) before starting with `biomod2`. As you saw in the previous image, the study area for this model cover most of Europe: would you suggest me to do something different with these data before using them in `biomod2?`
However, this is the plot you asked me that I obtained with this command:
```r
biomod2::plot(Capreolus_BIOMOD_data, calib.lines = calib_lines_allModels,
              plot.type = "points", plot.eval = T, do.plot = T)
``` 
![image](https://github.com/biomodhub/biomod2/assets/121942966/31346539-3aea-4041-a3d2-355127886ffd)

Regarding the pseudo-absences, would you suggest me also to try a different strategy to select them?

Thank you so much!!

**Comment by rpatin** [comment]
Hello Lorenzo,
Thank you for the update :pray: 

Concerning the dataset, thank you for sharing the info and the qgis and biomod2 points plots. There was likely an issue with the plot with `plot.type = 'raster'` although it is quite puzzling. The `biomod2` plot with all data is quite hard to read (it is hard to have a generic function to plot such diversity of dataset). However I think this may be part of the issue here. You are modeling the climatic niche of the species (given your resolution and the type of data you use). However your presence points cover most of Europe with some holes (which are likely to be linked with sampling effort). For the model, the presence can there be expected to be quite hard to distinguish from the background points (pseudo-absences). Indeed except northern Russia and a few islands (Ireland, Sardinia, Sicily) there are very few places without presences. The contrast between presences and background points will thus be quite limited and I believe this is what you observe with the relatively poor model performance. To be able to model the climatic niche for this species you would need to extend the area of interest to include areas of absences, e.g. Northern Africa, Near-East. Otherwise it will be hard for any model to predict the climatic niche. 

Concerning the question of the specific RF overfit, the response curves are a good tools. You can indeed check what were the response without changing the RF arguments. I would also check which variables are important (for RF but for other models as well) to focus your attention on the relevant response curves (the two or three most important variables across the algorithm). You can get variable importance plot with:
```
bm_PlotVarImpBoxplot(bm.out = myBiomodModelOut, group.by = c('expl.var', 'algo', 'algo'))
```
What you can see on the RF response curve is that it has a lot of very small variation, I think this was worse before you changed parameters and that would be the kind of thing that in your situation you would want to see disappear when you change the RF parameters.

Please let us know if something is unclear in the answer.
Best,
Rémi

**Comment by LorenzoBernicchi** [comment]
Dear @rpatin.
Thank you again for all the important advices!

I will try two things then:

- I will further increase the `nodesize` and `maxnodes` parameters in `randomForest`, to see if that will improve its performance;
- I will change my modeling area by adding some nearby regions, to see if the general performances of all algorithms will improve.

Thank you so much for now, I will update you with the results of these tries.

All the best,
Lorenzo
